zetaclient/chains/base/observer.go

```diff
@@ -406,6 +408,25 @@ func (ob *Observer) PostVoteInbound(
 		logs.FieldCoinType: coinType.String(),
 	}
 
+	cctxIndex := msg.Digest()
+
+	// The cctx is created after the inbound ballot is finalized
+	// 1. if the cctx already exists, we could try voting if the ballot is present
+	// 2. if the cctx exists but the ballot does not exist, we do not need to vote
+	_, err := ob.ZetacoreClient().GetCctxByHash(ctx, cctxIndex)
+	if err == nil {
+		// The cctx exists we should still vote if the ballot is present
+		_, ballotErr := ob.ZetacoreClient().GetBallotByID(ctx, cctxIndex)
```

Comment:
Can we skip this check entirely if the cctx is already finalized? Or do we want to try and vote on all cctx for auditing purposes

Reply:
cctx is created when enough votes comes in, so no m0re is needed in principle right?

Reply:
i'm thinking maybe just check cctx existence: 
exists -> no vote
non existing -> vote

wouldn't this be simpler?

Reply:
Yeah I think that make sense to me

Reply:
no we still need to allow votes if the ballot exists . The ballot gets finalized at 6/9 votes for mainnet for example , we should still accept the votes from the last 3 observers if the ballot is still present